python-avd/pyavd/_eos_designs/shared_utils/overlay.py

```diff
@@ -33,6 +33,8 @@ def vtep_loopback(self: SharedUtils) -> str:
     def evpn_role(self: SharedUtils) -> str | None:
         if self.underlay_router:
             default_evpn_role = self.node_type_key_data.default_evpn_role
+            if self.is_wan_router and self.inputs.wan_use_evpn_node_settings_for_lan:
+                default_evpn_role = None
```

Comment:
This logic will not handle the case where someone have copied the wan router type and set those variables themselves. Instead you need this at the top of the method (same for the others):
```suggestion
            if self.is_wan_router and not self.inputs.wan_use_evpn_node_settings_for_lan:
                return None
```

Reply:
Another option could be a special value of "wan_only" or something to use as a signal to other parts of the code. They often use != "none" which would still be true, but it would not match ibgp/ebgp?

Reply:
> This logic will not handle the case where someone have copied the wan router type and set those variables themselves. Instead you need this at the top of the method (same for the others):

I think my change is covering all use cases as well:

1. until now it was not possible to set the role to anything different from wan_role, so in theory no use can have something different than `wan_role == evpn_role`
2. if they enable the knob we use the default value of none and it can be overridden (which we want if they want to enable EVPN on LAN)

Assuming they have copied over the node type and don't set up the knob, the code should now be made to ignore the role anyways.

I am trying to think if your suggestion is breaking any of this.

Technically we have not changed the default in the documentation and I think returning `None` if the knob is not set for wan_router is changing the default value dynamically which ideally we should not do (?)

Reply:
The issues is:
- User creates a new node type, by copying from the defaults. This includes the evpn_role: server.
- This new code will respect that setting and it will trigger EVPN config on that device 
- Hence breaking change.

I suggest that we change the default to "none" in the schema and docs and ignore any other value unless the special knob has been set.
- Not breaking since nothing changes in output for the same inputs
- Even if the user copied and have `evpn_role: server` it would not matter.

Reply:
discussed on a call and agreed with your comment - will implement the change! thanks!